### PR TITLE
Changed DataTable to add headerGroups

### PR DESCRIPTION
**Pull request by ericsoderberghp**
#### What does this PR do?

Changed DataTable to add columnGroups

#### Where should the reviewer start?

doc.js

#### What testing has been done on this PR?

Added story and unit test.

#### How should this be manually tested?

story

#### What are the relevant issues?

https://github.com/grommet/grommet/issues/4445

#### Do the grommet docs need to be updated?

automatic

#### Should this PR be mentioned in the release notes?

yes

#### Is this change backwards compatible or is it a breaking change?

backwards compatible

### CODE REVIEW

File: src/js/components/DataTable/doc.js [hunk 1]
```diff
@@ -83,6 +83,17 @@ export const doc = DataTable => {
       `Cell border. You can set the border per context by passing an
       object with keys for 'heading', 'body', and/or 'footer'.`,
     ),
+    columnGroups: PropTypes.arrayOf(
```
Comment (ShimiSun) [inline src/js/components/DataTable/doc.js:86]: I'm thinking `groupHeader` or `headerGroups` will be a more straight forward naming.

**Comment by ericsoderberghp** [comment]
> Is there a way to style the new header?
> e.g. change the border or align the new header text 'Result' to the center?

There is now ;)

File: src/js/components/DataTable/Header.js [hunk 2]
```diff
@@ -39,6 +42,38 @@ const Header = ({
...
+            const headerGroup = headerGroups.groups.find(({ properties }) =>
+              properties.includes(property),
+            );
+            if (headerGroup) {
+              if (headerGroup.properties.indexOf(property) === 0) {
+                const { align, properties } = headerGroup;
+                return (
+                  <TableCell
+                    key={property}
+                    {...theme.dataTable.headerGroup}
```
Comment (ShimiSun) [inline src/js/components/DataTable/Header.js:58]: Should the styles be placed after the border/background/pad settings on lines 61-63? So the users will be able to override it from the theme.

File: src/js/components/DataTable/doc.js [hunk 3]
```diff
@@ -305,7 +321,12 @@ export const themeDoc = {
     defaultValue: "{ horizontal: 'small', vertical: 'xsmall' }",
   },
   'dataTable.header': {
-    description: 'Styles for the header.',
+    description: 'Styles for the header cells.',
+    type: 'object',
+    defaultValue: '{}',
+  },
+  'dataTable.headerGroup': {
+    description: 'Styles for the header group cells.',
```
Comment (ShimiSun) [inline src/js/components/DataTable/doc.js:329]: Is it Box styling? Table styling? we should be explicit about which styles the theme support.

**Comment by ShimiSun** [comment]
I'll be closing this PR until we decide to re-prioritize it.

**Pull request closed without merging**